### DISCUSSION TITLE
v1.20 beta1

### DISCUSSION THREAD

**Discussion by duracelltomi** [duracelltomi/gtm4wp#315] [General]
THE BIG CLEANUP RELEASE!

Lots of deprecated features removed.
The code of WooCommerce integration was cleaned and restructured, if you have any custom code that relies on GTM4WP internal data structure,
please update the plugin on a test version of your website before updating your live site!

* Removed deprecated Universal Analytics events for WooCommerce
* Removed deprecated filter GTM4WP_WPFILTER_AFTER_DATALAYER/gtm4wp_after_datalayer
* Removed deprecated filter GTM4WP_WPFILTER_ADDGLOBALVARS/gtm4wp_add_global_vars
* Removed: deprecated feature that puts the 'Do not track' option of the browser into the data layer. It is a deprecated browser feature therefore removed from this plugin.
* Removed: deprecated Google Optimize integration
* Removed: deprecated constants GTM4WP_OPTION_BLACKLIST_MACRO_* and GTM4WP_OPTION_BLACKLIST_<adsystemname>
* Removed: deprecated WordPress filter GTM4WP_WPFILTER_GETTHEGTMTAG (deprecated in v1.16)
* Changed: lots of internal function names and variable names changed, although is not recommended to depend on them, if you did, check your code
* Changed: renamed WP filter gtm4wp_datalayer_on_pageload to gtm4wp_woocommerce_datalayer_on_pageload to better reflect when it is called.
* Added: stockstatus key into the product array of every ecommerce action. Returns the value of WP_Product->get_stock_status(). Thanks [hans2103](https://github.com/hans2103).
* Added: integration with WebToffee GDPR Cookie Consent plugin. GTM4WP can not fire a GTM event when user consent changes or when a previously stored consent has been loaded.
* Fixed: add_payment_info and add_shipping_info events were not fired during checkout submit when not fired before on the page.

<hr /><em>This discussion was created from the release <a href='https://github.com/duracelltomi/gtm4wp/releases/tag/1.20-beta.1'>v1.20 beta1</a>.</em>

**Comment by hsiaoj** [comment]
Hello, We're testing the beta and the EE events are working well for us, thank you!  The 'Customer data in the dataLayer' option is enabled but we are not seeing it. Is that feature still available?

Best,
June Hsiao

**Reply by duracelltomi** [reply]
Yes, but it will only have values when there is a logged in customer or WP user. For guest checkouts, WooCommerce does not populate this data.

**Reply by hsiaoj** [reply]
Ah ok, got it.  Thanks for your prompt reply.
